pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1039,8 +1039,19 @@ async def on_package(pkg):
                 await self.put(dc)
 
             if settings.RPM_ITERATIVE_PARSING:
-                for pkg in parser.parse_packages_iterative(file_extension, skip_srpms=skip_srpms):
-                    await on_package(pkg)
+                if settings.RPM_USE_NEW_ITERATIVE_PARSER:
+
+                    @async_to_sync
+                    async def _package_callback(pkg):
+                        if not skip_srpms or pkg.arch != "src":
+                            await on_package(pkg)
```

Comment:
I don't like this, it's kind of hairy.  But I don't see how we could make it work otherwise since this API makes createrepo_c be in control of the execution - synchronously - but the code it needs to run is asynchronous.  The new API puts a barrier here where there wasn't before.

But I also see why it had to be this way, so we just have to deal with it I think.